msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java

```diff
@@ -161,5 +174,52 @@ private static boolean brokerInstalled(@NonNull final Context applicationContext
         return false;
     }
 
+    //This should be a background call..
+    private static void sayHelloToBroker(@NonNull final PublicClientApplicationConfiguration configuration)
+            throws ClientException {
+
+        IMicrosoftAuthService service;
+
+        final MicrosoftAuthClient client = new MicrosoftAuthClient(configuration.getAppContext());
+        final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
+
+        try {
+            service = authServiceFuture.get();
+            final Bundle requestBundle = new Bundle();
+            requestBundle.putString(
+                    AuthenticationConstants.Broker.CLIENT_ADVERTISED_MAXIMUM_BP_VERSION_KEY,
+                    AuthenticationConstants.Broker.BROKER_PROTOCOL_VERSION_CODE
+            );
 
+            if (!StringUtil.isEmpty(configuration.getRequiredBrokerProtocolVersion())) {
+                requestBundle.putString(
+                        AuthenticationConstants.Broker.CLIENT_CONFIGURED_MINIMUM_BP_VERSION_KEY,
+                        configuration.getRequiredBrokerProtocolVersion()
+                );
+            }
+
+            final Bundle resultBundle = service.hello(requestBundle);
+            if (null == resultBundle) {
+                throw new ClientException(
+                        ErrorStrings.BROKER_BIND_SERVICE_FAILED,
+                        "Unable to verify the broker app."
+                );
+            } else if (null != requestBundle.getString(AuthenticationConstants.OAuth2.ERROR)) {
+                throw new ClientException(
+                        requestBundle.getString(AuthenticationConstants.OAuth2.ERROR),
+                        resultBundle.getString(AuthenticationConstants.OAuth2.ERROR_DESCRIPTION)
+                );
+            }
+        } catch (RemoteException e) {
+            throw new ClientException(ErrorStrings.BROKER_BIND_SERVICE_FAILED,
+                    "Exception occurred while attempting to invoke remote service",
+                    e);
+        } catch (Exception e) {
+            throw new ClientException(ErrorStrings.BROKER_BIND_SERVICE_FAILED,
```

Comment:
Looks like we're using the same ErrorStrings for different error conditions. I'm wondering which of these occurrs as a result of power management being applied... and which may simply be transient.  I'm wondering if we need to consider some retries here.